module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -542,6 +542,44 @@ protected function fetchUserWithCql($query)
         return count($json->users) === 1 ? $json->users[0] : null;
     }
 
+    /**
+     * Helper function to retrieve paged results from FOLIO API
+     *
+     * @param string $responseKey Key containing values to collect in response
+     * @param string $interface   FOLIO api interface to call
+     * @param array  $query       CQL query
+     *
+     * @return array
+     */
+    protected function getPagedResults($responseKey, $interface, $query = [])
+    {
+        $count = 0;
+        $limit = 1000;
+        $offset = 0;
+
+        do {
+            $combinedQuery = array_merge($query, compact('offset', 'limit'));
+            $response = $this->makeRequest(
+                'GET',
+                $interface,
+                $combinedQuery
+            );
+            $json = json_decode($response->getBody());
+            if ($json == null) {
+                throw new ILSException('Could not get requests');
+            }
+            $total = $json->totalRecords ?? 0;
```

Comment:
Should this be a `+=` rather than an assignment?

Reply:
I think this is OK to be an assigment since we just want the total number of records we need to collect. In the response totalRecords will be the same on each page.

Reply:
Ahh, yes, you're right, I was misunderstanding the purpose of the field. Thanks!